User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs

```diff
@@ -0,0 +1,388 @@
+{
+    /// <summary>
+    /// Represents tests for CollectionSortHandler class.
+    /// </summary>
+    /// <owner>Anton Petrenko</owner>
+    [TestClass]
+    public class CollectionSortHandlerTest
+    {
+        /// <summary>
+        /// Collections the sort handler collection sort handler pass null value in constructor throw argument null exception.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_CollectionSortHandler_PassNullParserInConstructor_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            var reader = new StubDataReader(new StubIValidator());
+            var writer = new StubDataWriter<double>();
+
+            //
+            // Assert.
+            //
+            Assert.ThrowsException<ArgumentNullException>(() => new CollectionSortHandler<double>(reader, writer, null));
+        }
+
+        /// <summary>
+        /// Collections the sort handler collection sort handler pass null value in constructor throw argument null exception.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_CollectionSortHandler_PassNullReaderInConstructor_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            var parser = new StubIDataParser<double>();
+            var writer = new StubDataWriter<double>();
+
+            //
+            // Assert.
+            //
+            Assert.ThrowsException<ArgumentNullException>(() => new CollectionSortHandler<double>(null, writer, parser));
+        }
+
+        /// <summary>
+        /// Collections the sort handler collection sort handler pass null value in constructor throw argument null exception.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_CollectionSortHandler_PassNullWriterInConstructor_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            var parser = new StubIDataParser<double>();
+            var reader = new StubDataReader(new StubIValidator());
+
+            //
+            // Assert.
+            //
+            Assert.ThrowsException<ArgumentNullException>(() => new CollectionSortHandler<double>(reader, null, parser));
+        }
+
+        /// <summary>
+        /// Tests Execute if left empty unsorted collection it will throw exception.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_Execute_LeftEmptyUnsortedCollection_ThrowException()
+        {
+            using (ShimsContext.Create())
+            {
+                //
+                // Arrange.
+                //
+                var parser = new StubIDataParser<double>();
+                var reader = new ShimDataReader();
```

Comment:
When we use Shim, we need to create ShimContext:
ShimContext.Create().
Also don't forget to cover with using.